plugin.program.autowidget/resources/lib/common/utils.py

```diff
@@ -364,37 +377,46 @@ def _get_json_version():
     result = json.loads(call_jsonrpc(json.dumps(params)))['result']['version']
     return (result['major'], result['minor'], result['patch'])
 
-
-def get_files_list(path, titles=None):
-    if not titles:
-        titles = []
+def cache_files(path):
+    hash = hashlib.sha1(path).hexdigest()

```

Comment:
The cache is actually for a single path or url. A widget can contain many paths (either randomised by autowidget or just paged). In the new PR made a .history file and contains both the history of when the path content changed and which widgets this path was requested by (because in theory you could have multiple widgets with the same content). It's not going to make much sense to put references to the caches in the widget files though.